core-api/src/main/java/com/optimizely/ab/config/parser/AudienceJacksonDeserializer.java

```diff
@@ -76,5 +82,17 @@ private Condition parseConditions(List<Object> rawObjectList) {
 
         return condition;
     }
+
+    private Condition parseConditions(Object rawObject) {
+        List<Condition> conditions = new ArrayList<Condition>();
+
+        HashMap<String, String> conditionMap = (HashMap<String, String>)rawObject;
+        conditions.add(new UserAttribute(conditionMap.get("name"), conditionMap.get("type"),
```

Comment:
name and type are non null values.  you should probably test the get name or have a default value setup.

Reply:
In case where name or type is null then what should be the default value? 
currently we are getting illegal argument exception if we send null value in name or type.

Reply:
@mikeng13

Reply:
@mnoman09 @mikeng13 default can probably be empty string for now.  but, i'm confused.  were single leaf conditions allowed in the datafile after typed audiences or is this a fix for compat?

Reply:
here is the [Link](https://github.com/optimizely/fullstack-sdk-compatibility-suite/blob/6317c6a6b99e4bbcff84924a2c0e4a65233ea9b1/features/support/datafiles/audience_targeting.json#L1597) of line and datafile of compat suite in which we have single condition leaf root.  Due to these type of audience we were unable to parse datafile and all other testcases were failing which were not even related to this audience.